### PR TITLE
autotest: make more tests independent, use tmp_ fixtures, parametrize

### PR DESCRIPTION
**Pull request by dbaston**
## What does this PR do?

Resolves inter-test dependency issues in most files in `ogr` directory that do not involve a database. Will address those ones separately.

In my fork I am getting a segfault in the mingw build: https://github.com/dbaston/gdal/actions/runs/6075146965/job/16488720395#step:12:2192

```
Windows fatal exception: access violation
23: 
23: Current thread 0x00000a04 (most recent call first):
23:   File "D:/a/gdal/gdal/build/autotest/gcore/test_driver_metadata.py", line 396 in test_metadata_openoptionlist
```

I haven't modified this test but I also don't see the failure in `master`, so I'm not yet sure what to make of it.

## What are related issues/pull requests?

#4407

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

**Comment by rouault** [comment]
> In my fork I am getting a segfault in the mingw build: https://github.com/dbaston/gdal/actions/runs/6075146965/job/16488720395#step:12:2192
> 
> ```
> 
> ```

workaround (to disable it on mingw CI) added a few days ago per 84a5948bb3fe734a05638540db3284dc02cdba70

File: autotest/ogr/ogr_mitab.py [hunk 1]
```diff
@@ -594,127 +440,115 @@ def test_ogr_mitab_20():
...
-            time.sleep(1)
 
-        f = open("tmp/mitab_bounds.txt", "wb")
-        if i == 0:
-            f.write(
-                """Source = CoordSys Earth Projection 3, 33, "m", 3, 46.5, 44, 49, 700000, 6600000
+@pytest.mark.parametrize("i", (0, 1))
+def test_ogr_mitab_20bis(tmp_vsimem, tmp_path, i):
+
+    bounds_fname = str(tmp_path / "mitab_bounds.txt")
```
[the remark this answers is not in the document]
Comment (dbaston) [reply autotest/ogr/ogr_mitab.py]: Thanks for pointing this out. I updated in https://github.com/OSGeo/gdal/pull/8333/commits/4c0d37c23ebcea36f76ce01c5236dc962687e15f to explicitly test both the case where the timestamp has changed and where it is identical.

[the remark this answers is not in the document]
Comment (dbaston) [reply autotest/ogr/ogr_mitab.py]: Is it possible to see in Coveralls which lines lost coverage? I see under https://coveralls.io/builds/62456981 that some files still lost coverage, but I don't see a way to see which lines changed. Maybe this is not possible? https://github.com/lemurheavy/coveralls-public/issues/459

If not possible, I could try https://github.com/capgelka/lcov-diff locally.

[the remark this answers is not in the document]
Comment (rouault) [reply autotest/ogr/ogr_mitab.py]: > Is it possible to see in Coveralls which lines lost coverage?

I don't know...

I've seen weird changes of coverage in other PRs in files/areas unaffected by the change, which I suspect might be related to some network based tests sometimes skipped when the resource isn't available

[the remark this answers is not in the document]
Comment (dbaston) [reply autotest/ogr/ogr_mitab.py]: The `lcov-diff` tool does the job. I found one other error in my work on MITAB (https://github.com/OSGeo/gdal/pull/8348). Will make a pass through past PRs and see if anything else turns up.

**Pull request merged**